### PR TITLE
Check garbage collection on all plotting tests, allow specific skipping

### PR DESCRIPTION
**Pull request by MatthewFlamm**
### Overview

After #4500, `check_gc` fixture was only used in three test modules. It would be better to have it check all of the plotting tests, with optional skips.  This PR adds in this functionality by setting `check_gc` to be `autouse=True` in the `tests/plotting/conftest.py` and yielding an object which allows skipping the test.  A new fixture `skip_check_gc` can be used to skip specific tests (or modules) if warranted.

This PR is not aiming to fix any of the gc issues that might be uncovered. But specific uses of `skip_check_gc` will be an easier indication for bug fixing later.

### Details

First, this PR will push the changes that will expose failing tests, then skips will be added.

### CODE REVIEW

**Comment by banesullivan** [comment]
> After https://github.com/pyvista/pyvista/pull/4500, check_gc fixture was only used in three test modules.

I want to clarify that it was only used in those same three modules before that PR too. Just now there are more testing modules in that directory which didn't all pass the GC validation.

Thanks for implementing this! I couldn't find a clean way to make the future opt-out after setting `autouse=True`

File: tests/plotting/conftest.py [hunk 1]
```diff
@@ -36,22 +36,31 @@ def _is_vtk(obj):
...
-    yield
-
-    # Do not check for collection if the test session failed. Tests that fail
-    # also fail to cleanup their resources and this makes reading the unit test
-    # output more difficult.
-    #
-    # This applies to the entire session, so it's going to be the most useful
-    # when debugging tests with `pytest -x`
-    pyvista.close_all()
-    if request.session.testsfailed:
```
Comment (MatthewFlamm) [inline tests/plotting/conftest.py:52]: This removed block to skip gc test using when any previous test failed (using `request.session.testsfailed` was added in #2941.  I needed to turn this off to easily find all the failing tests here, and did not notice any confusing output in doing so.  @akaszynski implemented this and maybe can shed some light on whether it should still be included.

Comment (MatthewFlamm) [reply tests/plotting/conftest.py:52]: It seems that it my local pytest invocation is somehow different than the CI invocation and it does add confusing output for gc failures on the CI.  We might need to turn it back on.  Will do more digging.

Comment (MatthewFlamm) [reply tests/plotting/conftest.py:52]: I was misinterpreting my local vs. CI pytest results.  When looking closer, I did not see anything more complicated after removing this if block for the failures.  Note that in #4500 only a small number of tests were identified as failing (see https://github.com/pyvista/pyvista/pull/4500#issuecomment-1581261637), but in reality a large number fail (mostly from within the tests/plotting/jupyter folder, tests/plotting/test_axes, and tests/plotting/test_charts).  The discrepancy is due to this if block.

**Review by banesullivan — APPROVED** [review]
Awesome! All looks good to me

**Comment by MatthewFlamm** [comment]
Thanks, even though I'm wary of skipping `check_gc` when any other test has failed, I'd like to wait for comments on this before merging.

**Review by adeak — APPROVED** [review]
LGTM. I think it's the right call to always fail unless explicitly skipped, and following up with individual prodding of gc-failing tests. Thanks @MatthewFlamm.

**Pull request merged**